src/ngraph_mark_for_clustering.cc

```diff
@@ -213,6 +213,7 @@ Status MarkForClustering(Graph* graph) {
       type_constraint_map["Reshape"]["T"] = NGraphDTypes();
       type_constraint_map["Reshape"]["Tshape"] = NGraphIndexDTypes();
       type_constraint_map["Rsqrt"]["T"] = NGraphDTypes();
+      type_constraint_map["Shape"]["T"] = NGraphNumericDTypes();
```

Comment:
`T` is the attribute for the input type, so any nGraph-able datatype (`NGraphDTypes()`) should be fine.

In addition to this, `Shape` also has a type variable called `out_type`, so we also will need something like:

```
type_constraint_map["Shape"]["out_type"] = NGraphIndexDTypes();
```

(I think `NGraphIndexDTypes` includes int32 and int64, but I'm not sure.)

Reply:
Can the shape op take bool type? I wasn't sure so I used NGraphNumericDTypes.

And yes, NgraphIndexTypes includes int32 and int64.

Reply:
Yup, it can take any type as input.